modules/custom/openy_campaign/src/Plugin/Block/CampaignActivityStatisticsBlock.php

```diff
@@ -84,6 +84,8 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function build() {
+    // The block is rendered for each user separately.
+    // We can't cache it.
```

Comment:
we have user cache tags, or user context. https://www.drupal.org/docs/8/api/cache-api/cache-contexts
Let's work on this after release.